analyzers/tests/SonarAnalyzer.Test/TestCases/ExceptionsNeedStandardConstructors.cs

```diff
@@ -22,37 +22,37 @@ public class My_01_Exception : Exception // Noncompliant
         My_01_Exception(SerializationInfo info, StreamingContext context) {}
     }
 
-    public class My_02_Exception : Exception // Noncompliant
+    public class My_02_Exception : Exception // Compliant
     {
         public My_02_Exception() { }
 
         public My_02_Exception(string message) { }
 
         public My_02_Exception(string message, Exception innerException) { }
 
-        public My_02_Exception(SerializationInfo info, StreamingContext context) { }
+        public My_02_Exception(SerializationInfo info, StreamingContext context) { } // optional serialization constructor (should be protected)
     }
 
-    public class My_03_Exception : Exception // Noncompliant
+    public class My_03_Exception : Exception // Compliant
     {
         public My_03_Exception() { }
 
         public My_03_Exception(string message) { }
 
         public My_03_Exception(string message, Exception innerException) { }
 
-        private My_03_Exception(SerializationInfo info, StreamingContext context) { }
+        private My_03_Exception(SerializationInfo info, StreamingContext context) { }  // optional serialization constructor (should be protected)
```

Comment:
Educational: 
> All externally visible (public or protected) serialization ctors: .ctor(SerializationInfo, StreamingContext).

Means that this is not marked with the [Obsolete] warning? 
If so, should we continue raising S4027 on private/internal ctors?